src/bosh-director/spec/unit/jobs/delete_orphan_networks_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+module Bosh::Director
+  describe Jobs::DeleteOrphanNetworks do
+    let(:event_manager) { Api::EventManager.new(true) }
+    let(:task) { Bosh::Director::Models::Task.make(id: 42, username: 'user') }
+    let(:delete_orphan_networks_job) { instance_double(Bosh::Director::Jobs::DeleteOrphanNetworks, username: 'user', task_id: task.id, event_manager: event_manager) }
+
+    before { allow(Config).to receive(:current_job).and_return(delete_orphan_networks_job) }
+
+    describe '.enqueue' do
+      let(:job_queue) { instance_double(JobQueue) }
+
+      it 'enqueues a DeleteOrphanNetworks job' do
+        fake_orphan_networks_names = ['nw-1', 'nw-2']
+        Models::Network.make(name: 'nw-1', orphaned: true)
+        Models::Network.make(name: 'nw-2', orphaned: true)
+        expect(job_queue).to receive(:enqueue)
+          .with('fake-username', Jobs::DeleteOrphanNetworks, 'delete orphan networks', [fake_orphan_networks_names])
+        Jobs::DeleteOrphanNetworks.enqueue('fake-username', fake_orphan_networks_names, job_queue)
+      end
+
+      it 'errors if network is not orphaned' do
+        Models::Network.make(name: 'nw-2', orphaned: false)
+        expect do
+          Jobs::DeleteOrphanNetworks.enqueue(nil, ['nw-2'], JobQueue.new)
+        end.to raise_error(NetworkDeletingUnorphanedError)
+      end
+
+      it 'errors if network doesnot exist' do
+        Models::Network.make(name: 'nw-2', orphaned: false)
+        expect do
+          Jobs::DeleteOrphanNetworks.enqueue(nil, ['nw-4'], JobQueue.new)
+        end.to raise_error(NetworkNotFoundError)
+      end
+    end
+
+    describe '#perform' do
+      let(:event_log) { EventLog::Log.new }
+      let(:event_log_stage) { instance_double(Bosh::Director::EventLog::Stage) }
+      let(:cloud_factory) { instance_double(Bosh::Director::CloudFactory) }
+      let(:cloud) { instance_double(Bosh::Clouds::ExternalCpi) }
```

Comment:
We think it might make more sense if, in this file, instead of mocking out the CPI and testing the side effects of the DeleteOrphanNetworks, we instead mock out the `OrphanNetworkManager`, ensuring that it receive multiple calls to `delete_network`. The justification being that since the `DeleteOrphanNetworks` does not have a direct dependency on the CPI, it's unnecessary/not a good idea to test those side effects at this level

Reply:
@jrussett Thanks. I have modified the unit test to mock out the `OrphanNetworkManager` instead